hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/Utils.java

```diff
@@ -63,4 +65,20 @@ public static Optional<HederaKey> asHederaKey(@Nullable final com.hedera.hapi.no
             return Optional.empty();
         }
     }
+    /**
+     * Converts a set of {@link com.hedera.hapi.node.base.Key} to a set of {@link HederaKey}
+     *
+     * @param requiredNonPayerKeys
+     * @return
+     */
+    public static Set<HederaKey> asHederaKeys(Set<com.hedera.hapi.node.base.Key> requiredNonPayerKeys) {
+        if (requiredNonPayerKeys == null) {
+            return Set.of();
+        }
+        return requiredNonPayerKeys.stream()
```

Comment:
I believe this would filter `KeyList{}` from the input `Set<Key>` (again because `Utils.asHederaKey()` does validations). I.e. this _could_ e.g. change signing requirements for a transaction involving `0.0.800`. 🤔 

Probably doesn't matter given all this is temporary code, but worth noting.

Reply:
I think it will not filter, it will add an empty `KeyList` when converting to PBJ right ?

Reply:
Ah true, there should be an earlier `PreHandleException` in the case of an immutable account. Good point.